packages/web/src/pages/comment-history/components/desktop/CommentHistoryPage.tsx

```diff
@@ -0,0 +1,221 @@
+import { useCallback, useMemo, useRef } from 'react'
+
+import {
+  useGetCommentById,
+  useGetCommentsByUserId,
+  useGetCurrentUserId,
+  useGetTrackById,
+  useGetUserByHandle,
+  useGetUserById
+} from '@audius/common/api'
+import { Status } from '@audius/common/models'
+import {
+  Box,
+  Flex,
+  IconButton,
+  IconHeart,
+  LoadingSpinner,
+  Paper,
+  Skeleton,
+  Text,
+  TextLink
+} from '@audius/harmony'
+import { Comment, decodeHashId } from '@audius/sdk'
+import dayjs from 'dayjs'
+import InfiniteScroll from 'react-infinite-scroller'
+import { useNavigate } from 'react-router-dom-v5-compat'
+
+import { Avatar } from 'components/avatar'
+import { CommentBlockSkeletons } from 'components/comments/CommentSkeletons'
+import { Timestamp } from 'components/comments/Timestamp'
+import { Header } from 'components/header/desktop/Header'
+import { TrackLink, UserLink } from 'components/link'
+import Page from 'components/page/Page'
+import { useProfileParams } from 'pages/profile-page/useProfileParams'
+import { fullCommentHistoryPage } from 'utils/route'
+
+import { CommentText } from './CommentText'
+
+const messages = {
+  by: ' by ',
+  noComments: 'No comments',
+  view: 'View'
+}
+
+const UserComment = ({ commentId }: { commentId: number }) => {
+  const res = useGetCommentById(commentId)
+  const comment = res.data as Comment
+  const navigate = useNavigate()
+
+  const {
+    id,
+    entityId,
+    message,
+    createdAt,
+    userId,
+    isEdited,
+    mentions = [],
+    reactCount,
+    isCurrentUserReacted
+  } = comment
+
+  // @ts-expect-error - the id is a number
+  const { status } = useGetUserById({ id: userId })
+  const { data: track } = useGetTrackById({ id: decodeHashId(entityId) })
```

Comment:
can we decode in an adapter in the useCommentById hook?

Reply:
also use `useGetTrack`

Reply:
Similar to above, `HashId.parse` instead of `decodeHashId`.